From 72eecde2d89bcf7e7c21b03b87854eb9df7bfce5 Mon Sep 17 00:00:00 2001
From: Philipp Ammann <philipp.ammann@posteo.de>
Date: Tue, 11 Aug 2020 12:52:46 +0200
Subject: [PATCH 1/8] Move GTK2 checks behind MOZ_X11

GTK2 isn't compatible with Wayland and requires X11. No functional
change.

---
 old-configure.in | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/old-configure.in b/old-configure.in
index 2a386a5983f3..bb57cd2992c8 100644
--- a/old-configure.in
+++ b/old-configure.in
@@ -1642,14 +1642,12 @@ if test "$COMPILE_ENVIRONMENT"; then
     GLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_32
   fi
   if test "$MOZ_ENABLE_GTK"; then
-    if test "$MOZ_X11"; then
-      GDK_PACKAGES=gdk-x11-2.0
-    fi
     AC_DEFINE_UNQUOTED(GLIB_VERSION_MIN_REQUIRED,$GLIB_VERSION_MIN_REQUIRED)
     AC_DEFINE_UNQUOTED(GLIB_VERSION_MAX_ALLOWED,$GLIB_VERSION_MAX_ALLOWED)
-
-    PKG_CHECK_MODULES(MOZ_GTK2, gtk+-2.0 >= $GTK2_VERSION gtk+-unix-print-2.0 glib-2.0 >= $GLIB_VERSION gobject-2.0 gio-unix-2.0 $GDK_PACKAGES)
-    MOZ_GTK2_CFLAGS="-I${_topsrcdir}/widget/gtk/compat $MOZ_GTK2_CFLAGS"
+    if test "$MOZ_X11"; then
+      PKG_CHECK_MODULES(MOZ_GTK2, gtk+-2.0 >= $GTK2_VERSION gtk+-unix-print-2.0 glib-2.0 >= $GLIB_VERSION gobject-2.0 gio-unix-2.0 gdk-x11-2.0)
+      MOZ_GTK2_CFLAGS="-I${_topsrcdir}/widget/gtk/compat $MOZ_GTK2_CFLAGS"
+    fi
   fi
 fi # COMPILE_ENVIRONMENT
 
-- 
2.27.0
